src/mongo/crypto/crypto_openssl.cpp

```diff
@@ -39,25 +39,47 @@
 #include <openssl/hmac.h>
 #include <openssl/sha.h>
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#include <cstring>
+
+static void *OPENSSL_zalloc(size_t num)
```

Comment:
Because we're not using OPENSSL_zalloc elsewhere in this file, please inline it into EVP_MD_CTX_new. Thanks!